base/linalg/cholesky.jl

```diff
@@ -47,6 +44,7 @@ function CholeskyPivoted{T}(A::AbstractMatrix{T}, uplo::Char, piv::Vector{BlasIn
     CholeskyPivoted{T,typeof(A)}(A, uplo, piv, rank, tol, info)
 end
 
+RealSymOrHerm{S} = Union{Symmetric{<:Real,S}, Hermitian{<:Any,S}}
```

Comment:
I guess `RealHermSymComplexHerm{<:Real,<:StridedMatrix}` could be used for all methods in this file then? Worth adding a type alias for that here instead?